src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2373,6 +2471,13 @@ public static function get_tests() {
 			);
 		}
 
+		if ( 'production' === wp_get_environment_type() ) {
```

Comment:
Could we add an inline comment here, just to explain why the tests are only done in certain environments (I know it makes sense to us now, but it doesn't hurt to have the _why_ of it documented for the future).